lib/call-cabal-project-to-nix.nix

```diff
@@ -122,14 +122,35 @@ let
           else builtins.trace ("Using latest index state" + (if name == null then "" else " for " + name) + "!")
             (pkgs.lib.last (builtins.attrNames index-state-hashes));
 
+in
+  assert (if index-state-found == null
+    then throw "No index state passed and none found in ${cabalProjectFileName}" else true);
+
+let
+  # If a hash was not specified find a suitable cached index state to
+  # use that will contain all the packages we need.  By using the
+  # first one after the desired index-state we can avoid recalculating
+  # when new index-state-hashes are added.
+  # See https://github.com/input-output-hk/haskell.nix/issues/672
+  cached-index-state = if index-sha256 != null
+    then index-state-found
+    else
+      let
+        suitable-index-states =
+          builtins.filter
+            (s: s >= index-state-found)
```

Comment:
This deserves a note that it's probably a bit fragile.

Reply:
Added note and an assert to make sure index-state-found ends with `Z`